spec/test_track_spec.rb

```diff
@@ -68,37 +68,52 @@
   end
 
   describe "misconfguration_notifier" do
-    it "wraps a custom notifier in Wrapper" do
+    it "wraps a custom notifier in Wrapper without memoizing" do
       begin
-        default_notifier = TestTrack.misconfiguration_notifier
         fake_notifier = double
-        TestTrack.misconfiguration_notifier = fake_notifier
-        expect(TestTrack.misconfiguration_notifier.class).to eq TestTrack::MisconfigurationNotifier::Wrapper
+        fake_notifier_class = double(new: fake_notifier)
+        stub_const('FakeNotifier', fake_notifier_class)
+        TestTrack.misconfiguration_notifier_class_name = 'FakeNotifier'
+
+        expect(TestTrack.misconfiguration_notifier).to be_instance_of(TestTrack::MisconfigurationNotifier::Wrapper)
         expect(TestTrack.misconfiguration_notifier.underlying).to eq fake_notifier
       ensure
-        TestTrack.misconfiguration_notifier = default_notifier
+        TestTrack.instance_variable_set(:@misconfiguration_notifier_class_name, nil)
       end
     end
 
-    it "defaults to null notifier" do
-      expect(TestTrack.misconfiguration_notifier.class).to eq TestTrack::MisconfigurationNotifier::Wrapper
-      expect(TestTrack.misconfiguration_notifier.underlying.class).to eq TestTrack::MisconfigurationNotifier::Null
+    it "returns a new instance each time" do
+      begin
+        fake_notifier = double
+        fake_notifier_class = double
+        call_count = 0
+        allow(fake_notifier_class).to receive(:new) do
+          call_count += 1
+          fake_notifier
+        end
```

Comment:
we can't use the built-in rspec call count stuff for this?

Reply:
The validation calls it once, and I don't want to write a spec that says that two calls will result in 3 calls to the underlying instantiation.